mrmustard/physics/ansatze.py

```diff
@@ -393,9 +393,10 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         Returns:
             The tensor product of this ansatz and other.
         """
+
         As = [math.block_diag(a1, a2) for a1 in self.A for a2 in other.A]
         bs = [math.concat([b1, b2], axis=-1) for b1 in self.b for b2 in other.b]
-        cs = [math.outer(c1, c2) for c1 in self.c for c2 in other.c]
+        cs = [math.outer(c1, c2).reshape(-1) for c1 in self.c for c2 in other.c]
```

Comment:
If `c1` and `c2` can potentially be large arrays we shouldn't actually call `outer`, but somehow store them separately and use them as individual units (e.g. as part of a larger tensor contraction). We can leave it for now, but keep in mind that you will probably have to come back to this at some point.